File: prime_factor/prime_factor.py
import math


def prime_factor(num):
    """
    Finds all the prime factor of a number.

    :param num : number whose prime factors we want to find.
    :return : return a list of prime factors of number.
    """
    factor_list = []
    sqrt_num = int(math.sqrt(num)) + 1
    for start in range(2, sqrt_num):
        if num % start == 0:
            factor_list.append(start)
            while (num % start == 0):
                num /= start
    if num != 1:
        factor_list.append(num)
    return factor_list


def main():
    """
    Driver function
    """
    num = 362880
    print("Prime Factors are :", prime_factor(num))


if __name__ == '__main__':
    main()

File: prime_factor/prime_factor.go
package main

import (
	"fmt"
	"math"
)

// PrimeFactor finds all the prime factor of given number
func PrimeFactor(n int) []int {
	var primes []int
	sqrt := int(math.Sqrt(float64(n)))
	for i := 2; i <= sqrt; i++ {
		if n%i == 0 {
			primes = append(primes, i)
			for n%i == 0 {
				n = n / i
			}
		}
	}
	if n != 1 {
		primes = append(primes, n)
	}
	return primes
}

func main() {
	n := 8
	fmt.Println("Prime Factors are :")
	fmt.Println(PrimeFactor(n))
}
